In [1]:
import numpy as np
import matplotlib.pyplot as plt

In [2]:
l_onda  = 640e-9 
delta_onda = 10e-9

In [3]:
L = np.array([107.3e-2,84.0e-2,110.5e-2,76e-2])
delta_L = 0.05e-2*np.ones((1,4))

In [4]:
x2 = np.array([107e-2,86e-2,108.8e-2,75.4e-2])
error_x = 0.05e-2*np.ones((1,4))
x = x2/2

In [5]:
np.sin(x/np.sqrt(x**2+L**2))

array([0.43155195, 0.44006505, 0.42746248, 0.42990035])

In [6]:
d = (l_onda*np.sqrt(x**2+L**2))/x

In [7]:
d_o = (np.sqrt(x**2+L**2))/x
d_x = -L**2*l_onda/(x**2*np.sqrt(x**2+L**2))
d_L = l_onda*L/(x*np.sqrt(x**2+L**2))
print(d_o)
print(d_x)
print(d_L)

[2.24108486 2.19456529 2.26406196 2.25031415]
[-2.14713707e-06 -2.58812157e-06 -2.14397404e-06 -3.06577273e-06]
[1.07056695e-06 1.32487176e-06 1.05549491e-06 1.52078463e-06]


In [8]:
d_d = np.sqrt((d_o*delta_onda)**2+(d_x*error_x)**2+(d_L*delta_L)**2)

In [9]:
promedio = np.mean(d)
desviacion = np.std(d,ddof=1)

In [10]:
d

array([1.43429431e-06, 1.40452179e-06, 1.44899965e-06, 1.44020106e-06])

In [11]:
promedio 

1.4320042018556821e-06

In [12]:
error_d = np.mean(d_d)
print(promedio,error_d)

1.4320042018556821e-06 2.241922857203862e-08


\begin{equation}
    d = 1.43 \pm 0.02 \mu m
\end{equation}

\begin{align}
    d\sin{\phi} &= \lambda \\
    d &= \frac{\lambda}{\sin{\phi}} \\
    d &= \frac{\lambda}{\sin{(\arctan{x/L})}} \\
    d &= \frac{\lambda\sqrt{x^2+L^2}}{x}
\end{align}

In [13]:
A = (11.9e-2/2)**2*np.pi - (4e-2/2)**2*np.pi

In [14]:
d_A = np.sqrt(((11.9e-2/2)*np.pi*0.05e-2)**2+((4e-2/2)*np.pi*0.05e-2)**2)
print(A,d_A)

0.009865386330435346 9.860110133942768e-05


\begin{equation}
    A = 98.7 \pm 0.1 \,[cm]^2
\end{equation}

In [15]:
surco = (promedio/2)**2*np.pi
d_surco =  (promedio/2)*np.pi*error_d
print(surco,d_surco)

1.6105657750041584e-12 5.0429519959969636e-14


In [16]:
capacidad_en_bits = A/surco
error_capa_bits = np.sqrt((d_A/surco)**2+(-d_surco*A/surco**2)**2)
print(capacidad_en_bits,error_capa_bits)

6125416598.033616 201331025.53019363


In [17]:
capacidad_en_MB = capacidad_en_bits/8e6
error_capa_MB = error_capa_bits/8e6
print(capacidad_en_MB,error_capa_MB)

765.677074754202 25.1663781912742


In [18]:
LLL = 7.1e-2

In [19]:
x_a = 2.95e-2
x_v = 3.8e-2

In [20]:
def angulo(x,l):
    return x/np.sqrt(x**2+l**2)

In [21]:
agua = promedio*angulo(x_a,LLL)
vacio = promedio*angulo(x_v,LLL)

In [22]:
agua,vacio

(5.494480605805521e-07, 6.757295304691417e-07)

In [23]:
vacio/agua

1.2298333162831796